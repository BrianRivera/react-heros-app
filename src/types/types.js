export const types = {
    login: '[auth] login',
    logaut: '[auth] logaut',
}