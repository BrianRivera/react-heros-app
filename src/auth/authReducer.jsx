const { types } = require("../types/types");

const state = {
    name: 'Brina',
    logged: true
}

export const authReducer = (state = {}, action) => {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged:true
            }
        
            case types.logaut:
                return {
                    logged: false
                }
    
        default:
            return state;
            break;
    }
    
}
