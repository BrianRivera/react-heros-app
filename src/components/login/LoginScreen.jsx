import React, { useContext } from 'react'
import { authReducer } from '../../auth/authReducer'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({history}) => {
    
    //dispatch es una propedad del context
    const {dispatch} = useContext(AuthContext);
    
    
    const handleLogin = ()=>{

        const lastpath = localStorage.getItem('lastPath') || '/';

        dispatch({
            type: types.login,
            payload: {
                name: 'brian 2'
            }
        });
        history.replace(lastpath);   
    }

    return (
        <div className="container mt-5">
            <h1>login</h1>
            <hr/>

            <button
            className="btn btn-primary"
            onClick={handleLogin}
            >
                login
            </button>
        </div>
    )
}
