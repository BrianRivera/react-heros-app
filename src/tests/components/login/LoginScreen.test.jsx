import '@testing-library/jest-dom'
import React from 'react';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';


describe('Pruebas en login', () => {
    
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged:true,
            name:'brian'
        }
    }

    const history = {
        replace: jest.fn()
    }

    const wrapper = mount(
    <AuthContext.Provider value = {contextValue}>
        <MemoryRouter>
            <LoginScreen history={history}>
            </LoginScreen>
        </MemoryRouter>
    </AuthContext.Provider>
    )



    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de realizar el dispatch y la navegacion', () => {
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'brian 2'
            }
        });

        expect(history.replace).toHaveBeenCalledTimes(1);
        
        localStorage.setItem('lastPath','/dc');
        handleClick();
        expect(history.replace).toHaveBeenCalledWith('/dc')

    })
    
    

})
