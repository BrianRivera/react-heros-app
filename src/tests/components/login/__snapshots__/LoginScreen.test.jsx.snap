// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en login debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
